internal/consensus/reactor.go

```diff
@@ -46,8 +46,9 @@ type Reactor struct {
 	waitSync atomic.Bool
 	eventBus *types.EventBus
 
-	rsMtx cmtsync.Mutex
-	rs    *cstypes.RoundState
+	rsMtx         cmtsync.RWMutex
+	rs            *cstypes.RoundState
+	initialHeight int64 // under rsMtx
```

Comment:
Why do we need this value for every received message? Not questioning the change, but the rationale.

Reply:
We need it to be updated on every prune, since that can change our initial Height. I didn't see a synchronous point to update for prunes

Reply:
I think this is the first height of the blockchain, it never changes. In any case, the first block (height) stored is something we can get from the block store, not from the state.